src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/DataContent.cs

```diff
@@ -256,4 +259,142 @@ private string DebuggerDisplay
                 $"Data = {uri.Substring(0, MaxLength)}...";
         }
     }
+
+    /// <summary>
+    /// Loads a <see cref="DataContent"/> from a file path asynchronously.
+    /// </summary>
+    /// <param name="path">The file path to load the data from.</param>
+    /// <param name="mediaType">
+    /// The media type (also known as MIME type) represented by the content. If not provided,
+    /// it will be inferred from the file extension. If it cannot be inferred, "application/octet-stream" is used.
+    /// </param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests.</param>
+    /// <returns>A <see cref="DataContent"/> containing the file data with the inferred or specified media type and name.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="path"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="path"/> is empty or whitespace.</exception>
+    [Experimental("MEAI001")]
+    public static async Task<DataContent> LoadFromAsync(string path, string? mediaType = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNullOrWhitespace(path);
+
+        // Infer media type from the file extension if not provided
+        mediaType ??= MediaTypeMap.GetMediaType(path) ?? MediaTypeMap.DefaultMediaType;
```

Comment:
Eventually once MediaTypeMap is available in the TFMs we build for, we'll want to be referencing the official MediaTypeMap rather than the polyfill, and it doesn't have a DefaultMediaType. As such, we shouldn't be adding DefaultMediaType to the polyfill. Make it a private const on this class instead.